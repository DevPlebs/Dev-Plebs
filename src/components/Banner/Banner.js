import PropTypes from "prop-types"
import React from "react"

import {
  BannerWrapper,
  BannerContainer,
  Title,
  StyledLink,
  Navigation,
  NavigationItem,
} from "./styles"

const Banner = ({ children }) => (
  <BannerWrapper>
    <BannerContainer>
      <StyledLink to="/">
        <Title>devplebs.</Title>
      </StyledLink>
      <Navigation>
        <StyledLink to="/">
          <NavigationItem>Friday Night Deploys</NavigationItem>
        </StyledLink>
      </Navigation>
    </BannerContainer>
    {children}
  </BannerWrapper>
)

Banner.propTypes = {
  children: PropTypes.any,
}

export default Banner
