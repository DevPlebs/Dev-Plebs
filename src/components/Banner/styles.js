import tw from "tailwind.macro"
import styled from "@emotion/styled"
import bg from "../../images/hills.svg"

export const BannerWrapper = styled.header`
  ${tw`bg-no-repeat bg-top relative w-screen`}
  height: 28em;
  background-image: url(${bg});
`

export const Container = tw.div`flex justify-center h-full flex-col ml-20`

export const Title = styled.h1`
  ${tw`text-white font-bold ml-2`}
  font-size: calc(60px + (120 - 60) * ((100vw - 300px) / (1600 - 300)));
  text-shadow: 2px 2px 2px rgba(0, 0, 0, 0.6);
  font-family: "Rubik", sans-serif;
`

export const Subtitle = styled.h2`
  color: #ffdbb5;
  font-family: "Rubik", sans-serif;
  font-weight: 400;
  margin: -1rem 0 0 0.75rem;
  text-shadow: 1px 1px 1px rgba(30, 30, 30, 0.9);
`
