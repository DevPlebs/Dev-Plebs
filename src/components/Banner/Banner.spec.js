import React from "react"
import { create } from "react-test-renderer"

import Banner from "./Banner"

describe("Banner", () => {
  it("renders correctly", () => {
    const tree = create(<Banner />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
