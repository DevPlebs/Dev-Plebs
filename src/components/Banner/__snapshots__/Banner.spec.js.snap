// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly 1`] = `
.emotion-8 {
  background-repeat: no-repeat;
  background-position: bottom;
  background-size: cover;
  position: relative;
  height: 28em;
  background-image: url(hills.svg);
}

.emotion-7 {
  max-width: 64rem;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 {
  color: #fff;
  font-weight: 700;
  margin-left: 0.5rem;
  text-align: center;
  font-size: calc(60px + (120 - 60) * ((100vw - 300px) / (1600 - 300)));
  text-shadow: 2px 2px 2px rgba(0,0,0,0.8);
  font-family: "Rubik",sans-serif;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 1.5rem;
  margin-top: 0.5rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  width: 95vw;
  max-width: 64rem;
  margin-left: 1.5rem;
  margin-right: 1.5rem;
  border-bottom: 4px solid white;
  border-top: 4px solid white;
}

@media (min-width:640px) {
  .emotion-6 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-2 {
  color: #fff;
  text-align: center;
  font-size: 1.25rem;
  font-weight: 700;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  margin-left: 0;
  margin-right: 0;
  text-shadow: 1px 1px 1px rgba(30,30,30,0.9);
  list-style: none;
  font-family: "Rubik",sans-serif;
}

@media (min-width:640px) {
  .emotion-2 {
    margin-left: 1rem;
    margin-right: 1rem;
  }
}

@media (min-width:768px) {
  .emotion-2 {
    margin-top: 0;
    margin-bottom: 0;
  }
}

<header
  className="emotion-8"
>
  <div
    className="emotion-7"
  >
    <a
      className="emotion-1"
      href="/"
    >
      <h1
        className="emotion-0"
      >
        devplebs.
      </h1>
    </a>
    <ul
      className="emotion-6"
    >
      <a
        className="emotion-1"
        href="/friday-night-deploys"
      >
        <li
          className="emotion-2"
        >
          Friday Night Deploys
        </li>
      </a>
      <a
        className="emotion-1"
        href="/blog"
      >
        <li
          className="emotion-2"
        >
          Blog
        </li>
      </a>
    </ul>
  </div>
</header>
`;
